pkg/apis/core/types.go

```diff
@@ -6620,3 +6620,32 @@ type ImageVolumeSource struct {
 	// +optional
 	PullPolicy PullPolicy
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Resize is the query options to a Pod's resize call
+type Resize struct {
```

Comment:
A problem with these special-purpose request types is that admission controllers (and admission policy) only see this request type, and can't easily make decisions in the context of the full pod object. It also makes it more difficult to assert that there wasn't an intermediate change to the pod. In the KEP, I propose that the subresource type just be the full pod object. This is consistent with /status and /ephemeralContainers.